libeos-payg/efi.c

```diff
@@ -62,25 +63,26 @@ struct efi_ops {
 static struct efi_ops *efi;
 
 static gboolean
-clear_immutable (const char *name)
+clear_immutable (const char  *name,
+                 GError     **error)
 {
   unsigned int flags;
   int ret = 0;
   glnx_autofd int fd = -1;
 
   if (efi_fd == -1)
-    return FALSE;
+    return glnx_throw (error, "EFI ops not initialized");
 
   fd = openat (efi_fd, name, O_RDONLY);
```

Comment:
Am I wrong or should this throw here if the file isn't opened? Otherwise the `ioctl`s below would fail with `EBADF`, which I believe gets mapped to `G_IO_ERROR_FAILED`. Not that the callers are currently distinguishing errors, but IMO they should.

Reply:
Good catch! I spotted & fixed this issue in _read but not here.